src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,60 @@
-package core.basesyntax.service;
-
-import core.basesyntax.dao.StorageDao;
-import core.basesyntax.dao.StorageDaoImpl;
-import core.basesyntax.model.User;
-
-public class RegistrationServiceImpl implements RegistrationService {
-    private final StorageDao storageDao = new StorageDaoImpl();
-
-    @Override
-    public User register(User user) {
-        return null;
-    }
-}
+package core.basesyntax.service;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import core.basesyntax.service.exception.RegisterException;
+
+public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_VALID_LENGTH = 6;
+    private static final int MIN_VALID_AGE = 18;
+    private final StorageDao storageDao = new StorageDaoImpl();
+
+    @Override
+    public User register(User user) {
+        if (user == null) {
+            throw new RegisterException("Null-user can't be registered.");
+        }
+        userExists(user.getLogin());
+        loginIsValid(user.getLogin());
+        passwordIsValid(user.getPassword());
+        ageIsValid(user.getAge());
+        return storageDao.add(user);
+    }
+
+    private void userExists(String login) {
+        if (storageDao.get(login) != null) {
+            throw new RegisterException("User with login ["
+                    + login + "] already exists");
+        }
+    }
+
+    private void loginIsValid(String login) {

```

Comment:
```suggestion
    private void isLoginValid(String login) {
```
It’s name convention for methods that return a boolean (check other methods)